### ISSUE TITLE
How can I run a pipelined training job on 16 GPUs and 4 nodes in a cluster

### ISSUE THREAD

**Issue by monajalal** [Azure/azure-sdk-for-python#28778]
**Is your feature request related to a problem? Please describe.**

I have created a 4 node cluster and I am setting GPU ids to be 0..3. 

How can I either change train.py or my pipeline_run.py which uses `PythonScriptStep` with minimal changes so it can take advantage of all of the 4 nodes in my compute cluster and not just one of the nodes?
![Screenshot from 2023-02-13 09-10-05](https://user-images.githubusercontent.com/1892917/218481285-aa9617fe-7f59-40dc-bbc6-76e7e96e82dd.png)

Currently, each node only has 4 GPUs so I am hoping to get access to 16 GPUs to run my job.

A few lines of code for GPU setup in train.py:

```
parser.add_argument('--gpuids',
    nargs='+', 
    type=int, 
    default=[0,1,2,3], 
    help='GPUs to use')
net = torch.nn.DataParallel(net,device_ids=opt.gpuids).cuda()
```
in my `pipeline_run.py` I have:
```
train_cfg = ScriptRunConfig(source_directory=script_path,
                     script='train.py',
                     arguments=script_args,
                     compute_target=compute_target,
                     environment=env)

train_step = PythonScriptStep(
    name="train step",
    arguments=script_args,
    source_directory=train_cfg.source_directory,
    script_name=train_cfg.script,
    runconfig=train_cfg.run_config,
)
pipeline = Pipeline(ws, steps=[train_step])
run = exp.submit(pipeline)
```
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by catalinaperalta** [comment]
Hi @monajalal, could you please share the SDK library you are using? As well as the version? Thanks!

**Comment by monajalal** [comment]
Hi @catalinaperalta thanks a lot for your time.

Here is the libraries used in train.py
```
import mlflow
import mlflow.pytorch
```

from azureml.core.authentication import ServicePrincipalAuthentication

from azureml.core import Workspace



and the libraries used in pipeline_run.py are:

```
import os
import azureml.core

from azureml.data import OutputFileDatasetConfig
from azureml.core.compute import AmlCompute
from azureml.core.compute_target import ComputeTargetException
from azureml.pipeline.steps import PythonScriptStep
from azureml.pipeline.core import Pipeline

# You can find tenant id under azure active directory->properties

from azureml.core.authentication import ServicePrincipalAuthentication

from azureml.core import (
    Workspace,
    Experiment,
    Dataset,
    Datastore,
    ComputeTarget,
    Environment,
    ScriptRunConfig
)

```
The versions:

```
In [1]: import mlflow

In [2]: mlflow.__version__
Out[2]: '2.1.1'


$ pip show azureml-core
Name: azureml-core
Version: 1.48.0
Summary: Azure Machine Learning core packages, modules, and classes
Home-page: https://docs.microsoft.com/python/api/overview/azure/ml/?view=azure-ml-py
Author: Microsoft Corp
Author-email: 
License: https://aka.ms/azureml-sdk-license
Location: /anaconda/envs/jupyter_env/lib/python3.8/site-packages
Requires: adal, argcomplete, azure-common, azure-core, azure-graphrbac, azure-mgmt-authorization, azure-mgmt-containerregistry, azure-mgmt-keyvault, azure-mgmt-resource, azure-mgmt-storage, backports.tempfile, contextlib2, cryptography, docker, humanfriendly, jmespath, jsonpickle, knack, msal, msal-extensions, msrest, msrestazure, ndg-httpsclient, packaging, paramiko, pathspec, pkginfo, PyJWT, pyopenssl, python-dateutil, pytz, requests, SecretStorage, urllib3
Required-by: azureml-pipeline-core, azureml-telemetry, azureml-train-automl-client, azureml-train-core, azureml-widgets
```

Is this helpful for you?

Please note that I am intending to run my job as pipeline steps inside pipeline_run.py  from the submit_node which is a VM inside Azure to a GPU cluster with 4 nodes where each node has 4 GPUs not via jupyter notebook.



`pipeline = Pipeline(ws, steps=[train_step, inference_step])`

**Comment by catalinaperalta** [comment]
Thanks for the info @monajalal! Re-routing to the appropriate team to investigate.
